# Home Work 4

## Задача 1

О непрерывной равномерно распределенной случайной величине B известно, что ее дисперсия равна 0.2. Можно ли найти правую границу величины B и ее среднее значение зная, что левая граница равна 0.5? Если да, найдите их.

Импортируем необходимые библиотеки

In [1]:
import numpy as np
from scipy import stats

### Решение:

Дисперсия непрерывной равномерно распределенной случайной величины вычисляется по формуле:

$$D(X) = \dfrac{(b - a)^2}{12}.$$

Дисперсия $D(X)$ и левая граница $a$ извесны, выведем из формулы дисперсии формулу для расчёта правой грацицы $a$:

$$(b - a)^2 = D(X) \cdot 12$$

$$b = a + \sqrt{D(X) \cdot 12}$$

In [2]:
a = 0.5
D_x = 0.2
b = a + np.sqrt(D_x * 12)
b

2.049193338482967

Среднее значение непрерывной равномерно распределенной случайной величины равно математическому ожиданию и вычисляется по формуле:

$$M(X) = \dfrac{a + b}{2}$$

In [3]:
M_x = (a + b) / 2
M_x

1.2745966692414834

### Ответ:

In [4]:
print(f'\nПравая граница величины B = {b}\n\nСреднее значение = {M_x}')


Правая граница величины B = 2.049193338482967

Среднее значение = 1.2745966692414834


## Задача 2

Рост взрослого населения города X имеет нормальное распределение, причем, средний рост равен 174 см, а среднее квадратическое отклонение равно 8 см. Посчитайте, какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:
1. больше 182 см?
2. больше 190 см?
3. от 166 см до 190 см?
4. от 166 см до 182 см?
5. от 158 см до 190 см?
6. не выше 150 см или не ниже 190 см?
7. не выше 150 см или не ниже 198 см?
8. ниже 166 см?

Задачу можно решить двумя способами: без использования сторонних библиотек (numpy, scipy, pandas и пр.), а затем проверить себя с помощью встроенных функций

### Решение:

Так как рост взрослого населения города X имеет нормальное распределение, воспользуемся правилом 3-х сигм для вычисления вероятности.

Средний рост (математическое ожидание): $\mu = 174,$ cреднее квадратическое отклонение: $\sigma = 8$.

По правилу 3-х сигм: 

- 99,7% всех значений лежат в пределе: $$\mu \pm 3\sigma = 174 \pm 24,$$ 
- 95,4% — в пределах $$\mu \pm 2\sigma = 174 \pm 16,$$ 
- 68,3% значений лежат в пределах $$\mu \pm \sigma = 174 \pm 8.$$ 

### Ответ:

Вероятность того, случайным образом выбранный взрослый человек имеет рост:
- больше 182 см: **0,1585**
- больше 190 см: **0.023**
- от 166 см до 190 см: **0.954**
- от 166 см до 182 см: **0.819**
- от 158 см до 190 см: **0.997**
- не выше 150 см или не ниже 190 см: **0.0245**
- не выше 150 см или не ниже 198 см: **0.003**
- ниже 166 см: **0.1585**

## Задача 3

Коробки с шоколадом упаковываются автоматически. Их средняя масса равна 1.06 кг. Известно, что 5% коробок имеют массу, меньшую 1 кг. Найдите: а) среднее квадратическое отклонение массы коробки, б) процент коробок, имеющих массу больше 1.1 кг.

### Решение:

In [5]:
loc = 1.06
precision = 0.00001
p = 0
scale = 0
while p < 0.05:
    scale += precision
    p = stats.norm.cdf(1, loc, scale)
print(scale, p)

0.0364800000000005 0.05001204583680406


Получили что вес коробок представляет собой нормально распределённую случайную величину с параметрами $a = 1.06, \sigma = 0.0365$. Найдём долю коробок с массой более 1.1 кг.

In [6]:
p = 1 - stats.norm.cdf(1.1, loc, scale)
print(p)

0.13643193002222642


### Ответ:

а) среднее квадратическое отклонение массы коробки: $\sigma = 0.0365$

б) процент коробок, имеющих массу больше 1.1 кг: $p = 13.65\%$